lib/storage/data/external/AwsClient.js

```diff
@@ -116,7 +125,8 @@ class AwsClient {
         // we call data.put to create a delete marker, but it's actually a
         // delete request in call to AWS
         if (keyContext.isDeleteMarker) {
-            return this._client.deleteObject(params, putCb);
+            return this._client.send(new DeleteObjectCommand(params)).then(res => putCb(null, res))
+            .catch(err => putCb(err));
```

Comment:
```suggestion
            .catch(putCb);
```
Can double check other occurrences in this file & elsewhere